src/compiler/transformers/module/module.ts

```diff
@@ -871,12 +888,13 @@ namespace ts {
             const left = node.left;
             // If the left-hand-side of the binaryExpression is an identifier and its is export through export Specifier
             if (isIdentifier(left) && isAssignmentOperator(node.operatorToken.kind)) {
-               const sourceFile = getSourceFileOfNode(node);
-               const sourceFileName = sourceFile ? sourceFile.fileName : undefined;
-                if (bindingNameExportSpecifiers && bindingNameExportSpecifiers[sourceFileName] && hasProperty(bindingNameExportSpecifiers[sourceFileName], left.text)) {
+                if (bindingNameExportSpecifiersForFileMap &&
+                    bindingNameExportSpecifiersForFileMap[currentSourceFileIdDuringSubstitution] &&

```

Comment:
Can we store the result of this lookup in a local variable so we aren't performing the same lookup twice in a row?